models/Version/Adapter/VersionStorageAdapterInterface.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Commercial License (PCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Model\Version\Adapter;
+
+interface VersionStorageAdapterInterface
+{
+    /**
+     * @param int|null $metaDataSize
+     * @param int|null $binaryDataSize
+     * @return string
+     */
+    public function getStorageType(int $metaDataSize = null,
+                                   int  $binaryDataSize = null): string;
+
+    /**
+     * @param int $id
+     * @param int $cId
+     * @param string $cType
+     * @param string $metaData
+     * @param string $storageType
+     * @param mixed|null $binaryDataStream
+     * @param string|null $binaryFileHash
+     * @param int|null $binaryFileId
+     * @return void
+     */
+    public function save(int $id,
+                         int $cId,
+                         string $cType,
+                         string $storageType,
+                         string $metaData,
+                         mixed $binaryDataStream = null,
+                         string $binaryFileHash = null,
+                         int $binaryFileId = null) : void;
```

Comment:
```suggestion
    public function save(Version $version) : void;
```
@mcop1 I'm wonderning whether we should change the signatures of all the methods to just use `Version`, to make it less extensive, or was there a particular reason to specify all the individual parameters?

Reply:
@brush just wanted to keep most of the pimcore dependencies and logic separated from the storage adapter. If we pass the `Version` object, we have to set some properties inside the save method of the storageadapter or double check if the data is an asset:
https://github.com/pimcore/pimcore/pull/11458/files#diff-440c14d8343f7c3272fb27e0ec1577e5178eb73eef4f165f0ead5f0326794973R239

On the other hand just passing `Version` would be less extensive indeed.

Reply:
Yep understand 😊 But since those 2 things are anyway deeply connected, I'd go for `Version` instead and maybe have some helper methods in a trait or so if it makes sense. @fashxp @dvesh3 wdyt?

Reply:
I would really use the property params if this interface is or will be used by other types. If this is specific to `version` then i don't see harm in passing the object itself.

Reply:
I'd say it's completely specific to `Version` right? I don't see any other use for it 🤔